src/components/Menus/MenuItem.js

```diff
@@ -0,0 +1,207 @@
+    onClick         : PropTypes.func,
+    /** Callback function fired when the menu item is focused. */
+    onFocus         : PropTypes.func,
+    /** Used by menu to keep track of current focus index. */
+    onMenuItemFocus : PropTypes.func,
+    /** Whether or not to prevent default when menu item is clicked */
+    preventDefault     : PropTypes.bool,
+    /** Role HTML attribute */
+    role            : PropTypes.string,
+    /** Customize style of MenuItem */
+    style           : PropTypes.shape({}),
+    /** Override tabIndex property*/
+    tabIndex        : PropTypes.number,
+    /** Whether or not to use use the tabIndex HTML attribute */
+    useTabIndex     : PropTypes.bool,
+    /** Underlying value. Also, passed into onClick function */
+    value           : PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+      PropTypes.bool
+    ]).isRequired,
+  }
+
+  static defaultProps = {
+    disabled: false,
+    role: 'menuitem',
+    useTabIndex: true,
+    focus: false,
+    preventDefault: true
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { focus } = this.props
+    if (!focus && nextProps.focus) {
+      setTimeout( () => {
+        this.menuItem.focus()
+      }, 0)
+    }
+  }
+
+  handleClick = (e) => {
+    const {
+      disabled,
+      index,
+      onClick,
+      value,
+      label,
+      preventDefault
+    } = this.props
+    preventDefault && e.preventDefault()
+
+    const option = {value: value, label: label}
+    !disabled && onClick && onClick(e, option, index)
+  }
+
+  handleKeyDown = (event) => {
+    switch(event.key) {
+      case 'Enter':
+        this.props.preventDefault && event.preventDefault()
+        this.handleClick()
+    }
+  }
+
+  handleFocus = (event) => {
+    const { onFocus, onMenuItemFocus, index } = this.props
+    onMenuItemFocus && onMenuItemFocus(index)
+    onFocus && onFocus(index, event)
+  }
+
+  renderLeftIcon() {
+    const { leftIcon, leftIconStyles } = this.props
+    if (!leftIcon) { return }
+
+    const iconComponent = typeof leftIcon === 'string'
+      ? <Icon name={leftIcon} style={[styles.leftIconStyles, leftIconStyles]} />
+      : leftIcon
+
+    return (
+      <div
+        style={styles.iconContainer}
+      >
+        <div style={{display: 'flex'}}>
+          { iconComponent }
+        </div>
+      </div>
+    )
+  }
+
+  renderMenuItem() {
+    const {
+      children,
+      label,
+      labelStyles,
+    } = this.props
+
+    if (children) {
+      return children
+    }
+
+    return (
+      <div style={styles.menuitem}>
+        { this.renderLeftIcon() }
+        <div style={[styles.label, labelStyles]}>
+          { label }
+        </div>
+      </div>
+    )
+  }
+
+  render() {
+    const {
+      disabled,
+      role,
+      style,
+      tabIndex,
+      useTabIndex
+    } = this.props
+
+    return (
+      <div
+        ref={(node) => this.menuItem = node}
+        role={role}
```

Comment:
I would be against this implementation. We should not be using a div. Please take a look at these two examples:

http://pauljadam.com/demos/rwd-aria-menubar.html
http://pauljadam.com/demos/aria-menubar.html